sandbox/modules/custom-codecs/src/main/java/org/opensearch/index/codec/customcodec/Lucene92CustomCodec.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodec;
+
+import org.apache.lucene.codecs.*;
+import org.apache.lucene.codecs.lucene90.*;
+import org.apache.lucene.codecs.lucene92.Lucene92HnswVectorsFormat;
+import org.apache.lucene.codecs.perfield.PerFieldDocValuesFormat;
+import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
+import org.apache.lucene.codecs.perfield.PerFieldPostingsFormat;
+
+public class Lucene92CustomCodec extends Codec {
```

Comment:
Apologies, just a question, the hierarchy changed from `FilterCodec` to `Codec`, why is that? (it seems like `FilterCodec` took care of delegation).

Reply:
Was re-writing the code and forgot that it does delegation. Fixed now.